rasa/core/processor.py

```diff
@@ -580,9 +582,91 @@ def _log_action_on_tracker(self, tracker, action_name, events, policy, confidenc
             e.timestamp = time.time()
             tracker.update(e, self.domain)
 
-    def _get_tracker(self, sender_id: Text) -> Optional[DialogueStateTracker]:
+    @staticmethod
+    def _session_start_timestamp_for(tracker: DialogueStateTracker) -> Optional[float]:
+        """Retrieve timestamp of the beginning of the last session start for
+        `tracker`.
+
+        Args:
+            tracker: Tracker to inspect.
+
+        Returns:
+            Timestamp of last `SessionStarted` event if available, else timestamp of
+            oldest event. `None` if no events are available.
+
+        """
+        if not tracker.events:
+            return None
```

Comment:
how about returning the current time and saving the `optional` handling this way?